lib/vanagon/platform/windows.rb

```diff
@@ -34,6 +34,143 @@ def package_name(project)
         "#{project.name}-#{project.version}-#{project.release}-#{@architecture}.msi"
       end
 
+      def add_repository(definition)
+        definition = URI.parse(definition)
+
+        ps1_command = install_chocolately_ps1 (definition)
+
+
+        commands = ["cat <<EOF; #{ps1_command}; EOF > install-chocolatey.ps1"]
+
+        #install_chocolately_on_host(definition)
+        puts "Now doing Add_Repo #{definition}"
+
+        if definition.scheme =~ /^(http|ftp)/
+          # use choco to install the package.
+          # choco source add -n=bob -s "https://somewhere/out/there/api/v2/"
+          puts "Adding Repo #{definition}"
+          commands << "echo choco add -s #{definition}"
+        end
+
+        commands
+      end
+
+      # Batch up powershell commands to extract out and install chocolatey
+      #
+      # @param command [String] optional repo target for built packages defined
+      #   at the project level
+      # @return [String] Wrapped Powershell Command
+      def install_chocolately_ps1(url)
+        batch_contents = <<-BATCH
```

Comment:
Vanagon has the `[String#undent](https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/extensions/string.rb)` extension method, so this HEREDOC doesn't need to be strictly aligned to column 1. Usage would look like this:

```ruby
  batch_contents = <<-BATCH.undent
    some big long string, which will have all leading whitespace to the length of the first line stripped out
  BATCH
```